### ISSUE TITLE
Is there a way to change rename order when exist duplicate names?

### ISSUE THREAD

**Issue by supperchong** [rollup/rollup#3975]
### Feature Use Case
//a.js
```js
function add(a,b){
    return a+b
}
export {add}
```

//main.js
```js
import * as a from './a'
function add(){
    return  a.add(1,2)
}
console.log(add())
```

// bundle out,js
```js
function add(a,b){
    return a+b
}

function add$1(){ // the name changed
    return  add(1,2)
}
console.log(add$1());
//# sourceMappingURL=out.js.map
```
When there are duplicate names, it will rename the variable from inner module to outer.
Is there a way to reverse the rename order? or add a custom config to support that?
// expect out.js
```js
function add$1(a,b){ //expect to chang the name
    return a+b
}

function add(){  
    return  add$1(1,2)
}
console.log(add());
//# sourceMappingURL=out.js.map
```

**Comment by lukastaegert** [comment]
This is not possible and I am not sure it is useful to make it configurable as this may raise expectations that names could be predictable, which I want to avoid. But if I understand you correctly, you think it is better that e.g. names in entry points should be retained as opposed names in dependencies. So from this perspective, we could just change the order in which we deconflict variables to use reverse execution order instead of execution order. Do you agree?

**Comment by supperchong** [comment]
@lukastaegert yeah. Thanks for reply. That's just what I need. I want to retain names in entry points as it may do different things with the same name in dependencies. 

> we could just change the order in which we deconflict variables to use reverse execution order instead of execution order

Could you give an example?

**Comment by lukastaegert** [comment]
I created a fix in ##3977. This is the situation without the fix:

https://rollupjs.org/repl/?version=2.39.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGZGVwLmpzJyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMCdtYWluJyUzQiU1Q25jb25zb2xlLmxvZyhmb28pJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyZGVwLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwZm9vJTIwJTNEJTIwJ2RlcCclM0IlNUNuY29uc29sZS5sb2coZm9vKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

And this is what you get using the fix:

https://rollupjs.org/repl/?circleci=14379&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGZGVwLmpzJyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMCdtYWluJyUzQiU1Q25jb25zb2xlLmxvZyhmb28pJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyZGVwLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwZm9vJTIwJTNEJTIwJ2RlcCclM0IlNUNuY29uc29sZS5sb2coZm9vKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by supperchong** [comment]
Perfect, thanks very much!
